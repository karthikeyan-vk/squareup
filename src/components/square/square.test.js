import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SquareComponent from './square';
import Square from '../../game/square';

test('renders correctly', () => {
  const square = new Square(0, 0);
  const component = renderer
    .create(<SquareComponent square={square} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

test('renders with number', () => {
  const square = new Square(0, 0);
  square.setNumber(5);
  const component = renderer
    .create(<SquareComponent square={square} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

test('renders current square', () => {
  const square = new Square(0, 0);
  const component = renderer
    .create(<SquareComponent square={square} current />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

test('renders destination square', () => {
  const square = new Square(0, 0);
  const component = renderer
    .create(<SquareComponent square={square} destination />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

test('calls next on click', () => {
  const mockNext = jest.fn();
  const square = new Square(0, 0);
  const component = shallow(<SquareComponent square={square} next={mockNext} />);
  component.simulate('click');
  expect(mockNext.mock.calls.length).toBe(1);
  expect(mockNext.mock.calls[0][0]).toBe(square);
});
