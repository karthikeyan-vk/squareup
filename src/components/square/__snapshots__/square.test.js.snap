// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="square empty"
  onClick={[Function]}
>
  <span
    className="number"
  />
</div>
`;

exports[`renders current square 1`] = `
<div
  className="square empty current"
  onClick={[Function]}
>
  <span
    className="number"
  />
</div>
`;

exports[`renders destination square 1`] = `
<div
  className="square empty destination"
  onClick={[Function]}
>
  <span
    className="number"
  />
</div>
`;

exports[`renders with number 1`] = `
<div
  className="square"
  onClick={[Function]}
>
  <span
    className="number"
  >
    5
  </span>
</div>
`;
